From 0e7834d498fbe55c11e1f1956e6b0e735e337c7c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Aida=20Jonikien=C4=97?= <aidas957@gmail.com>
Date: Sat, 10 Aug 2024 19:23:27 +0300
Subject: [PATCH] winewayland: Add missing default surface fallback in
 WindowPosChanging().

This fixes a blank window content issue that happens even with basic Wine programs.

Fixes: b9879d5adc1 ("winewayland: Remove now unnecessary WindowPosChanging checks.")
---
 dlls/winewayland.drv/window.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dlls/winewayland.drv/window.c b/dlls/winewayland.drv/window.c
index 7c694246fb7..815f60808c7 100644
--- a/dlls/winewayland.drv/window.c
+++ b/dlls/winewayland.drv/window.c
@@ -431,15 +431,18 @@ BOOL WAYLAND_WindowPosChanging(HWND hwnd, UINT swp_flags, BOOL shaped, const REC
                                const RECT *client_rect, RECT *visible_rect)
 {
     struct wayland_win_data *data = wayland_win_data_get(hwnd);
+    BOOL ret;
 
     TRACE("hwnd %p, swp_flags %04x, shaped %u, window_rect %s, client_rect %s, visible_rect %s\n",
           hwnd, swp_flags, shaped, wine_dbgstr_rect(window_rect), wine_dbgstr_rect(client_rect),
           wine_dbgstr_rect(visible_rect));
 
     if (!data && !(data = wayland_win_data_create(hwnd, window_rect, client_rect, visible_rect))) return FALSE; /* use default surface */
+
+    ret = !!data->wayland_surface; /* use default surface if we don't have a window */
     wayland_win_data_release(data);
 
-    return TRUE;
+    return ret;
 }
 
 
-- 
GitLab

